### ISSUE TITLE
Replace the background service with a wrapper script

### ISSUE THREAD

**Issue by edmorley** [memcachier/memcachier-tls-buildpack#11]
The background service has a number of disadvantages:
- Errors don't appear in the app's log output sent to the Heroku logging system.
- If the stunnel service dies, then Heroku won't know to restart it (unlike if the main dyno process dies).
- It's a bit magic, since other than the buildpack appearing in `heroku buildpacks`, there's no indication of interaction with another process, when developers are inspecting the `Procfile`.

The [heroku-buildpack-pgbouncer](https://github.com/heroku/heroku-buildpack-pgbouncer) buildpack instead uses a wrapper script that is put in the `Procfile`, which resolves most of the above. We could probably use much of their implementation, which is pretty well maintained (https://github.com/heroku/heroku-buildpack-pgbouncer/commits/master) rather than reinventing the wheel.

Note: Whilst this is similar to #1, that was mainly about auto-rewriting environment variables, which is independent of switching from background service to wrapper script (they could be rewritten using a `/app/.profile.d/` script instead).

**Comment by edmorley** [comment]
Or alternatively we can borrow from:
https://github.com/heroku/heroku-buildpack-redis

**Comment by dterei** [comment]
This seems like a good way forward. Sorry that we've been letting this languish, we'll try to find time but would also happily accept PR :)

**Comment by edmorley** [comment]
I've looked at the implementations in heroku-buildpack-pgbouncer and heroku-buildpack-redis - they are very similar, though:
* ignoring the missing pgbouncer parts, heroku-buildpack-redis seems to be a newer fork of that implementation
* heroku-buildpack-pgbouncer is likely soon removing the stunnel parts, since pgbouncer now supports it natively (heroku/heroku-buildpack-pgbouncer#68)

As such, I think copying heroku-buildpack-redis is the best bet.

There were a few issues with heroku-buildpack-redis that I've opened PRs for to reduce the amount of churn later. Once they are resolved I'll double back to this.
